### PR TITLE
Deal with held messages, deal with empty queue

### PR DESCRIPTION
**Pull request by bernhardschmidt**
Thanks for your work, it is immensely useful. I've added a few small fixes, see below.

### CODE REVIEW

**Comment by alexjurkiewicz** [comment]
Looks great! I've added you as a committer, please feel free to merge this

**Comment by antoniovl** [comment]
Any plans to merge this PR anytime soon? :-)

**Comment by bernhardschmidt** [comment]
Hrm, I cannot merge it.

@alexjurkiewicz Could you check the permissions again and/or merge yourself?

File: apq.py
Comment (alexjurkiewicz) [inline apq.py:41]: Is this comment still valid?

**Review by alexjurkiewicz — CHANGES_REQUESTED** [review]
I've had a bit of time to review this PR. Please have a look and let me know.

BTW, once this PR is merged, I plan to archive this repo, so if you prefer feel free to close this and pre-emptively fork.

File: apq.py
Comment (alexjurkiewicz) [inline apq.py:63]: This should be part of the test on line 39.

**Pull request closed without merging**
